target/linux/ath79/dts/ar7161_buffalo_wzr-hp-ag300h.dts

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar7100.dtsi"
+
+/ {
+	compatible = "buffalo,wzr-hp-ag300h", "qca,ar7161";
+	model = "Buffalo WZR-HP-AG300H/WZR-600DHP";
+
+	aliases {
+		led-boot = &diag;
+		led-failsafe = &diag;
+		led-upgrade = &diag;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	extosc: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-output-names = "ref";
+		clock-frequency = <40000000>;
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		diag: diag {
+			label = "buffalo:red:diag";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+	};
+
+	ath9k-leds {
+		compatible = "gpio-leds";
+		band2g_a {
+			label = "buffalo:amber:band2g";
+			gpios = <&ath9k0 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		usb {
+			label = "buffalo:green:usb";
+			gpios = <&ath9k0 3 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			trigger-sources = <&usb_ochi_port>, <&usb_echi_port>;
+			linux,default-trigger = "usbport";
+		};
+
+		band2g_g {
+			label = "buffalo:green:band2g";
+			gpios = <&ath9k0 5 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			linux,default-trigger = "phy0tpt";
+		};
+
+		band5g_g {
+			label = "buffalo:green:band5g";
+			gpios = <&ath9k1 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			linux,default-trigger = "phy1tpt";
+		};
+
+		router {
```

Comment:
These also seem to be wrong. You mixed up gpio controllers.

Reply:
You mean the router led? Same as for the USB LED, I ported it like this from ar71xx, and it does control the "Router" LED correctly.

Reply:
Yeah, now I see... But it's really not a conventional way of connecting leds.

Reply:
I hope you mean "not conventional by Buffalo" :) Is there any way to do this better in the device tree?

Reply:
It is correct if they really connected the LEDs there :-)